temporallib/auth/auth.py

```diff
@@ -14,51 +14,76 @@
 import google.auth.transport.requests
 from typing import Union
 from dataclasses import asdict
+import os
 
 
-@dataclass
 class MacaroonAuthOptions:
     """
     Defines the parameters for authenticating with Candid.
     """
 
-    macaroon_url: str
-    username: str
-    keys: KeyPair
+    macaroon_url: str = None
```

Comment:
nitpick: If they can stay `None` then it makes them `Optional[type]` I think.

Reply:
I tried setting it to `Optional[type]` but it complained about the `__init__` method missing required positional arguments for some reason.

Reply:
Interesting, I don't know why that would happen off the top of my head. Actually, I believe optional attributes need to be the last attributes in the class, could it be that?